src/Catalog/CatalogCommitItem.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// Represents a single item in a catalog commit.
+    /// </summary>
+    public sealed class CatalogCommitItem : IComparable
+    {
+        private const string _typeKeyword = "@type";
+
+        private CatalogCommitItem(
+            Uri uri,
+            string commitId,
+            DateTime commitTimeStamp,
+            IReadOnlyList<string> types,
+            IReadOnlyList<Uri> typeUris,
+            PackageIdentity packageIdentity)
+        {
+            Uri = uri;
+            CommitId = commitId;
+            CommitTimeStamp = commitTimeStamp;
+            PackageIdentity = packageIdentity;
+            Types = types;
+            TypeUris = typeUris;
+        }
+
+        public Uri Uri { get; }
+        public DateTime CommitTimeStamp { get; }
+        public string CommitId { get; }
+        public PackageIdentity PackageIdentity { get; }
+        public IReadOnlyList<string> Types { get; }
+        public IReadOnlyList<Uri> TypeUris { get; }
+
+        public int CompareTo(object obj)
+        {
+            return CommitTimeStamp.CompareTo(((CatalogCommitItem)obj).CommitTimeStamp);
```

Comment:
Same comment as above. Public methods should validate their arguments before attempting such casts.